apps/meteor/app/models/server/raw/LivechatInquiry.ts

```diff
@@ -34,4 +34,49 @@ export class LivechatInquiryRaw extends BaseRaw<ILivechatInquiryRecord> {
 	async setLastMessageByRoomId(rid: string, message: IMessage): Promise<UpdateWriteOpResult> {
 		return this.updateOne({ rid }, { $set: { lastMessage: message } });
 	}
+
+	async findNextAndLock(queue?: string): Promise<ILivechatInquiryRecord | undefined> {
+		const date = new Date();
+		const result = await this.col.findOneAndUpdate(
+			{
+				status: LivechatInquiryStatus.QUEUED,
+				...(queue && { queue }),
+				$or: [
+					{
+						locked: true,
+						lockedAt: {
```

Comment:
Since we're querying based on these fields, I think we should add them as indexes